### ISSUE TITLE
os.link is implemented to make symlinks instead of hardlinks

### ISSUE THREAD

**Issue by sruggier** [jmcgeheeiv/pyfakefs#75]
I was trying to use the atomicwrites package under pyfakefs, and after adding fsync support (see #73), the next issue I noticed was that the fake version of os.link is aliased to the fake os.symlink, which has very different behaviour in cases where the original file is renamed or deleted:
```
  # TODO: Link doesn't behave like os.link, this needs to be fixed properly.
  link = symlink
```

**Comment by jmcgeheeiv** [comment]
Excellent observation.  It would be fabulous if you could issue a pull request for this, too.

**Comment by sruggier** [comment]
Yeah, I've been working on it on and off. I have a working implementation that I'll make a PR out sometime in the next few days.

**Comment by jmcgeheeiv** [comment]
Considering this fixed by #95 and #97.
